ugs-core/src/com/willwinder/universalgcodesender/uielements/panels/MachineStatusPanel.java

```diff
@@ -121,116 +120,156 @@ public MachineStatusPanel(BackendAPI backend) {
-        workPanel.add(machinePositionXValue, "span 3, al right, wrap");
-        workPanel.add(resetYButton);
-        workPanel.add(workPositionYLabel, "al right");
-        workPanel.add(workPositionYValue, "growx, bottom");
-        workPanel.add(machinePositionYValue, "span 3, al right, wrap");
-        workPanel.add(resetZButton);
-        workPanel.add(workPositionZLabel, "al right");
-        workPanel.add(workPositionZValue, "growx, bottom");
-        workPanel.add(machinePositionZValue, "span 3, al right, wrap");
-        add(workPanel,"growx, span 2");
-
-        // Enabled pin reporting.
-        pinStatusPanel.setLayout(new MigLayout("flowy, wrap 3"));
-        pinStatusPanel.add(pinX);
-        pinX.setEnabled(false);
-        pinStatusPanel.add(pinY);
-        pinY.setEnabled(false);
-        pinStatusPanel.add(pinZ);
-        pinZ.setEnabled(false);
-        pinStatusPanel.add(pinProbe);
-        pinProbe.setEnabled(false);
-        pinStatusPanel.add(pinDoor);
-        pinDoor.setEnabled(false);
-        pinStatusPanel.add(pinHold);
-        pinHold.setEnabled(false);
-        pinStatusPanel.add(pinSoftReset);
-        pinSoftReset.setEnabled(false);
-        pinStatusPanel.add(pinCycleStart);
-        pinCycleStart.setEnabled(false);
+    private void addAxisPanel(char axis, JLabel work, JLabel machine) {
+        RoundedPanel axisPanel = new RoundedPanel(COMMON_RADIUS);
+        axisPanel.setBackground(ThemeColors.VERY_DARK_GREY);
+        axisPanel.setForeground(ThemeColors.LIGHT_BLUE);
+        axisPanel.setLayout(new MigLayout("fillx, wrap 2, inset 7, gap 0", "[left][grow, right]"));
+
+        RoundedPanel resetPanel = new RoundedPanel(COMMON_RADIUS);
+        resetPanel.setForeground(ThemeColors.LIGHT_BLUE);
+        resetPanel.setBackground(ThemeColors.DARK_BLUE_GREY);
+        resetPanel.setHoverBackground(ThemeColors.MED_BLUE_GREY);
+        resetPanel.setLayout(new MigLayout("inset 10 20 10 20"));
+        JLabel axisLabel = new JLabel(String.valueOf(axis));
+        axisLabel.setForeground(ThemeColors.LIGHT_BLUE);
+        resetPanel.add(axisLabel, "al center, dock center, id axis");
+        JLabel zeroLabel = new JLabel("0");
+        zeroLabel.setForeground(ThemeColors.LIGHT_BLUE);
+        resetPanel.add(zeroLabel, "pos (axis.x + axis.w - 4) (axis.y + axis.h - 20)");
+
+        work.setForeground(ThemeColors.LIGHT_BLUE);
+        machine.setForeground(ThemeColors.LIGHT_BLUE);
+        axisPanel.add(resetPanel, "sy 2");
+        axisPanel.add(work);
+        axisPanel.add(machine, "span 2");
+
+        fontManager.addAxisResetLabel(axisLabel);
+        fontManager.addAxisResetZeroLabel(zeroLabel);
+        fontManager.addWorkCoordinateLabel(work);
+        fontManager.addMachineCoordinateLabel(machine);
+
+        add(axisPanel,"growx, span 2");
+
+        resetPanel.addClickListener(() -> resetCoordinateButton(axis) );
+        axisResetControls.add(resetPanel);
+    }
+
+    private Timer createTimer() {
```

Comment:
The `ControllerListener` is the deprecated way to get these events, and `ControllerStatusListener` is the new version to get it. Did you switch to this method for performance reasons?

Reply:
Assuming you commented the wrong code. I'll respond to the next comment on this issue.

Reply:
Marking as done for tracking my tasks.